content/docs/infrastructure-wg.md

```diff
@@ -0,0 +1,196 @@
+  principles thereof: What should the development workflow and the 
+  infrastructure look like?
+* Identify the need for addons for Github the community is missing right now 
+  (e.g. subscribing to labels on Github Issues).
+* Come to a decision on using Phabricator vs. GitHub Pull Requests for code
+  reviews, then implement the change.
+* Create a roadmap for the handover of the existing infrastructure to a 
+  contractor. Components ready for immediate takeover:
+  * Phabricator (in case we stay with it)
+  * Pre-merge testing
+  * Buildbot workers
+  * clangd remote index for LLVM
+  * [LNT service](http://lnt.llvm.org/)
+  * ...
+* Create new design and structure for the LLVM website, migrate the existing 
+  content there.
+* Create a test strategy for the LLVM project and refactor the existing build 
+  infrastructure to meet the new test strategy. Some ideas:
+  * Multi-stage CI: run “cheap” builds first, run “expensive” builds only on
+    commits where the cheap builds have passed.
+  * Set up a collector dashboard showing build results for all CI systems per 
+    commit, e.g. Treeherder or something GitHub based.
+  * Investigate if we can get a good deal from GitHub Actions with sponsored 
+    hardware.
+  * Create a plan of the "important" buildbot configuration we need in order to 
+    detect the bugs with large impact (contributors and downstream). Then have
+    one central team/contractor operate these to ensure consistency and quality.
+* Gather user feedback on the existing infrastructure: Where are the pain 
+  points? What would our community like to see improved?
+* Migration of all python scripts and tools to Python 3 as Python 2 is 
+  deprecated.
+* Reach out to downstream infrastructure teams (e.g. Rust, Tensorflow) and 
+  understand if we have infrastructure topics we want to cooperate on.
+* Define Service Level Objectives (SLO) for our infrastructure and monitor 
+  them. Set up monitoring and altering for our infrastructure.  Derive measure
+  where we do not meet our SLOs.
+* Migrate Windows setup instructions off GnuWin32 as it’s deprecated and add
+  installation instructions users can copy-and-paste into their shell.
+* Investigate the Rust project infrastructure and see what would be helpful to 
+  LLVM. Some examples:
+  * Create a benchmark tool to compare the performance of the compiler and the
+    compiled code across versions, there is also a project for that.
+  * Set up some central source of truth for managing user accounts, group
+    membership and user/group permissions. Then configure all tools 
+    accordingly.
+* Investigate if/where we need to have Terms of Service and Pricacy Agreements.
+* Figure out where we need to be able to handle GDPR-related requests and
+  how that could be done.
+* Decide what to do with the mailing lists and GDPR deletion requests.
+* Decide if we want to have mandatory pre-merge testing and what it should 
```

Comment:
I know that this is not the place to comment on this topic but in analogy you can probably only say that there needs to be lunch break but you cannot and should not tell what to eat.

Reply:
Not sure if I get your analogy. I guess we need to work out a tangible proposal (maybe as video 😉 ) and discuss that with the community. That proposal must explain:
* the workflows for the user
* the checks we want to run
* the expected latency of the checks
* overriding failed checks in case of some type of emergency
* ...